examples/benchmarks/contextvalues_test.go

```diff
@@ -105,19 +105,20 @@ func useContext(ctx context.Context, j, mod, v int) {
 	}
 }
 
-const expectedOutput = `{"logger":"","level":0,"msg":"ping","string":"hello world","int":1,"float":1,"i":1,"j":2}`
+const expectedOutput = `{"logger":"","level":0,"msg":"ping","i":1,"j":2,"string":"hello world","int":1,"float":1}`
 
 func discardWithV(b *testing.B, v int) logr.Logger {
 	logger := funcr.NewJSON(func(actualOutput string) {
 		if actualOutput != expectedOutput {
 			b.Fatalf("expected %s, got %s", expectedOutput, actualOutput)
 		}
 		calls++
-	}, funcr.Options{})
-	logger = logger.WithContextValues(
-		logr.ContextKey{Key: contextKey1{}, Name: "i"},
-		logr.ContextKey{Key: contextKey2{}, Name: "j"},
-	)
+	}, funcr.Options{
```

Comment:
I dislike that you can only set this at a place where you know the impl - you can't ever modify it later.  Am I reading that right?  I think that disqualifies this approach

Reply:
That depends. We have taken the approach before that `main` (or it's equivalent helper functions) is where logging gets configured. When both the tracing library and the logging library are completely decoupled, then that is the only place where the logger can be configured to log the trace and span IDs, because only `main` knows about both. Then it's not a problem that a custom API of the logger backend has to be used.

We could extend `ContextLogSink` and add an API there for setting or adding `ContextKey` if we think that a standardized API would be needed.

Reply:
Agree that "main is where logging gets configured".  What I don't know is whether "extract these values from context" is something main knows about and can set for the whole program, or whether that set might change in different subsystems.

Things like "do you want to log file:line" make sense in that mode - I am not sure about context.  My immediate reaction was that it is more localized, but I admit I have not used this pattern, so...tell me I am wrong?

Reply:
For the trace + span ID I expect that this will get configured by the program. Before we tag a new logr release and thus commit to the new API, we should verify that this use case works by doing a prototype PR for Kubernetes.

But I see value in allowing localized extensions of the ContextKeys, therefore I prefer the solution with `Logger.WithContextValues`.

Reply:
Do you want to remove the other commit from this series so we can solicit more feedback?

Reply:
Yes, as we seem to agree on the preferred solution now I'll do that.